src/hooks/useOutsideClickDetector.ts

```diff
@@ -11,9 +15,13 @@ const useOutsideClickDetector = (
   ref: React.RefObject<HTMLElement>,
   onClickOutsideDetected: () => void,
   enableDetection: boolean,
+  options?: Options,
 ) => {
   useEffect(() => {
     const handleClickOutside = (event) => {
+      // if maxWidth is specified and clientWidth is greater than maxWidth, do not trigger `onClickOutsideDetected`
+      if (options?.maxWidth && document.documentElement.clientWidth > options.maxWidth) return;
```

Comment:
I have mixed feelings regarding this. I think it would be a better approach to leverage `enableDetection` instead of adding `options` because the logic of checking the width doesn't belong to the hook in my opinion so we can move this logic to `src/components/QuranReader/SidebarNavigation/SidebarNavigation.tsx` and instead of setting the value of `enableDetection` to `true`, we set the value to `document.documentElement.clientWidth > 768`